lib/model/model.go

```diff
@@ -94,21 +94,23 @@ type Model struct {
 	clientName    string
 	clientVersion string
 
+	fmut               sync.RWMutex                                           // protects the below
```

Comment:
(Placement changed because Go best practices is that the mutex lives on top of the things it protects...)